### PR TITLE
Kspviswa photo upload api take2 - Add photo upload API #13

### PR DESCRIPTION
**Pull request by kspviswa**
Add photo upload API #13
===

This is follow-up pull request for addressing review comment on pull request https://github.com/google/personfinder/pull/109.

Since my branch was already cluttered with lots of stale branches & my master is way behind Google's PF repo, I decided to do a clean-up and start fresh. This commit addresses the following review comments.

* Added `self.auth.domain_write_permission` constraint for photo upload
* Revamped the logic with better exception handling
* Added constraint for uploaded file size < 10 MB
* Modified the code as not to use same variable names
* Provided necessary error messages with relevant information
* Provided XML escaping with `django.utils.html.escape` utility

Kindly review the pull request & accept the merge if deemed fit.

*PS : I had completed unit testing for the use-cases*

### CODE REVIEW

File: app/api.py [hunk 1]
```diff
@@ -341,6 +343,42 @@ def get(self):
...
+    def post(self):
+        if not (self.auth and self.auth.domain_write_permission):
+            self.info(
+                403,
+                message='Missing or invalid authorization key',
+                style='plain')
+            return
+
+        # Check for empty body
+        if len(self.request.body) < 1:
```
Comment (gimite) [inline app/api.py]: Can you write "if not self.request.body" instead? It is preferred per style guide we follow:
https://google-styleguide.googlecode.com/svn/trunk/pyguide.html#True/False_evaluations

File: app/api.py [hunk 2]
```diff
@@ -341,6 +343,42 @@ def get(self):
...
+            self.error(400, "Request body must not be empty")
```
Comment (gimite) [inline app/api.py]: I believe you need "return" after self.error to abort the method. Same below.

File: app/api.py [hunk 3]
```diff
@@ -341,6 +343,42 @@ def get(self):
...
+
+        # Size check for uploaded file
+        if len(self.request.body) > PHOTO_UPLOAD_MAX_SIZE:
+            self.error(400, "Size of uploaded file is greater than 10MB")
+
+        try:
+            photoImg = images.Image(self.request.body)
```
Comment (gimite) [inline app/api.py]: Can you rename this to photo_img instead, for consistency of variable name?

**Comment by gimite** [comment]
Sorry for late. Thanks for applying my comments! Just a few more comments. It would be nice if you push the follow-up changes to the same branch, so I can see it in this pull request.

**Comment by kspviswa** [comment]
@gimite 

Thanks for the review comments. I had updated the code addressing your review comments. Please review the same.

**Comment by kspviswa** [comment]
@gimite 

When I tried to push the changes for bug #123, this pull request also got updated. I think I screwed the branching again. Could you please cherry-pick only the relevant changes. Sorry for the inconvenience.

**Comment by gimite** [comment]
Looks good to me. Merged to the master branch. Sorry for delay, and thanks for the contribution!

**Comment by kspviswa** [comment]
Thanks for merging the change. I just have few queries.

1) I see a label "Closed with unmerged commits" . Does this mean that, my pull request is not accepted fully?
2) Incidentally, I don't see this commit tagged under my name, under "Commits" page in my name. https://github.com/google/personfinder/commits?author=kspviswa .

Could you please explain me about this anomaly.

**Comment by gimite** [comment]
> 1) I see a label "Closed with unmerged commits" . Does this mean that, my pull request is not accepted fully?

Right. I didn't fully merge your branch because your branch also contains changes for bug #123 , as you explained above. I excluded a commit for that.

> 2) Incidentally, I don't see this commit tagged under my name, under "Commits" page in my name. https://github.com/google/personfinder/commits?author=kspviswa .

Hm I do see your two changes in the page above:
https://github.com/google/personfinder/commit/0900d03451e359ef5355de23c7f92e338273b52d
https://github.com/google/personfinder/commit/2f655b65485ebe2fe95cd820c1c6745e2fa7e547

**Comment by kspviswa** [comment]
@gimite .

Thank you. I understood the mistake. I will do clean branching next time :-) .
BTW, can this issue https://github.com/google/personfinder/issues/13 be closed?

**Comment by gimite** [comment]
Yes, just closed.

**Pull request closed without merging**